hydroflow/src/compiled/pull.rs

```diff
@@ -11,6 +11,82 @@ where
 {
     state: L::Repr,
 }
+pub struct StreamJoinState<K, L>
+where
+    L: LatticeRepr,
+{
+    tab: HashMap<K, L::Repr>,
+}
+
+impl<K, L> Default for StreamJoinState<K, L>
+where
+    L: LatticeRepr,
+{
+    fn default() -> Self {
+        Self {
+            tab: HashMap::new(),
```

Comment:
I'm kind of curious why the state doesn't have buffers as in SymmetricHashJoin. I would have expected this to look like "half of a SymmetricHashJoin".

Reply:
The buffer in SymmetricHashJoin is needed because an incoming value might result in multiple output values, but we can only output one value at a time. Instead of buffering, here we output all values together as a single lattice point, I think.